modules/formatting/formatting_tips.py

```diff
@@ -0,0 +1,81 @@
+from utils.utils import blockquote
+from modules.error.friendly_error import FriendlyError
+from discord.ext import commands
+
+
+class FormattingTips:
+	def __init__(self):
+		# maps format name to pair containing preview markdown and escaped markdown
+		self.formats = {
+			"italics": ("*italics*", "\*italics* or \_italics_"),
+			"bold": ("**bold text**", "\**bold text**"),
+			"underline": ("__underline__", "\__underline__"),
+			"strikethrough": ("~~strikethrough~~", "\~~strikethrough~~"),
+			"spoiler": ("||spoiler|| (click to reveal)", "\||spoiler||"),
+			"inline": ("`inline code`", "\`inline code`"),
+			"codeblock": (
+				'```cpp\ncout << "Code Block";\n```',
+				'\```cpp\n// replace this with your code\ncout << "Code Block";\n```',
+			),
+		}
+		# alternative ways to request formats (spaces and case is already ignored)
+		self.aliases = {
+			"italic": "italics",
+			"inlinecode": "inline",
+			"spoilers": "spoiler",
+			"underlines": "underline",
+			"code": "codeblock",
+			"codeblocks": "codeblock",
+			"codesnippet": "codeblock",
+			"snippet": "codeblock",
+			"strike": "strikethrough",
+		}
+
+	def all_markdown_tips(self) -> str:
+		"""return a list of all markdown tips"""
+		message = "**__Markdown Tips__**\n\n"
+		for format in self.formats:
+			(preview, escaped) = self.formats[format]
+			message += f"{preview}\n"
+			message += f"{blockquote(escaped)}\n\n"
+		return message
+
+	def individual_info(self, ctx: commands.Context, format: str) -> str:
+		"""return info for given format"""
+		normalized = self.__normalize(ctx, format)
+		if normalized == "codeblock":
```

Comment:
there could be other block type formats, such as quotes. maybe each tip can have a property is_block, and check for that instead of checking for specifically "codeblock"